examples/plot_compare_reduction.py

```diff
@@ -73,3 +90,29 @@
 plt.ylim((0, 1))
 plt.legend(loc='upper left')
 plt.show()
+
+###############################################################################
+# Illustration of ``CachedPipeline`` instead of ``Pipeline``
+###############################################################################
+# It is sometimes interesting to store the state of a specific transformer
+# since it could be used again. Using a pipeline in ``GridSearchCV`` triggers
+# such situations. Therefore, we replace ``Pipeline`` with ``CachedPipeline``
+# to memoize the transfomers within the pipeline.
+#
+# .. warning::
+#     Note that this example is, however, only an illustration since for this
+#     specific case fitting PCA is not necessarily slower than loading the
+#     cache. Hence, use ``CachedPipeline`` when the fitting of a transformer
+#     is costly.
+
+# Create a temporary folder to store the transformers of the pipeline
+cachedir = mkdtemp()
+memory = Memory(cachedir=cachedir, verbose=10)
+cached_pipe = CachedPipeline([('reduce_dim', PCA()),
+                              ('classify', LinearSVC())],
+                             memory=memory)
+
+# This time, a cached pipeline will be used within the grid search
+grid = GridSearchCV(cached_pipe, cv=3, n_jobs=2, param_grid=param_grid)
+digits = load_digits()
+grid.fit(digits.data, digits.target)
```

Comment:
Maybe a comment here (in a notebook-style cell rendered by sphinx-gallery) on how this is faster and cooler. Just to conclude the example.